force-app/main/default/classes/HouseholdsApp.cls

```diff
@@ -58,57 +58,45 @@ public inherited sharing class HouseholdsApp {
     }
 
     public void onBeforeInsertOf(Accounts accounts) {
-        if (isAddressManagementEnabled()) {
+        if (AddressService.isAddressManagementEnabled()) {
             normalizeLineBreaksFor(accounts.householdAccounts());
         }
-        if (isOrgAccountAddressesEnabled()) {
+        if (AddressService.isOrgAccountAddressesEnabled()) {
             normalizeLineBreaksFor(accounts.orgAccounts());
         }
     }
 
     public void onAfterInsertOf(Accounts accounts, TDTM_Runnable.DmlWrapper dmlWrapper) {
-        createNewAddressesAndAddToDmlWrapperForProcessing(accounts, dmlWrapper);
+        createAddressesFromNewAccounts(accounts, dmlWrapper);
     }
 
-    private void createNewAddressesAndAddToDmlWrapperForProcessing(Accounts accounts, TDTM_Runnable.DmlWrapper dmlWrapper) {
-        List<Account> listAccCreateAddr = new List<Account>();
-        if (isAddressManagementEnabled()) {
-            listAccCreateAddr.addAll(accounts.withAddresses().householdAccounts());
-        }
-        if (isOrgAccountAddressesEnabled()) {
-            listAccCreateAddr.addAll(accounts.withAddresses().orgAccounts());
-        }
-        // create any new Address objects
-        if (listAccCreateAddr.size() > 0) {
-            createAddrFromAcc(listAccCreateAddr, dmlWrapper);
+    private void createAddressesFromNewAccounts(Accounts accounts, TDTM_Runnable.DmlWrapper dmlWrapper) {
+        Accounts accountsWithNewAddressesToCreate = accounts.newAccountsWithAddressesToCreate();
+        if (accountsWithNewAddressesToCreate.getRecords().size() > 0) {
+            AddressService.createAddressesFor(accountsWithNewAddressesToCreate, dmlWrapper, false);
         }
     }
 
-    /*******************************************************************************************************
-    * @description for each Account, create a new default address and add it to dmlWrapper
-    * @param listAcc a list of Accounts
-    * @param dmlWrapper to hold the Addresses that need creating
-    ********************************************************************************************************/
-    private void createAddrFromAcc(List<Account> listAcc,
-            TDTM_Runnable.DmlWrapper dmlWrapper) {
-        List<Address__c> listAddr = new List<Address__c>();
-        for (Account acc : listAcc) {
-            NPSP_Address npspAddress = new NPSP_Address(acc);
-            listAddr.add(npspAddress.getRecord());
+    private List<Account> getNewAccountsWithAddressesToCreate(Accounts accounts) {
```

Comment:
This is basically duplicated by Accounts.newAccountsWithAddressesToCreate().  Unless this was created to coordinate with the Household refactoring, shouldn't one of them go away?

Reply:
good call. This method wasn't actually being used anyway